lib/msf/core/post/file.rb

```diff
@@ -537,19 +537,29 @@ def rm_rf(*remote_dirs)
   alias :dir_rm :rm_rf
 
   #
-  # Rename a remote file.
+  # Renames a remote file and returns true on success and false
+  # on failure
   #
   # @param old_file [String] Remote file name to move
   # @param new_file [String] The new name for the remote file
   def rename_file(old_file, new_file)
+  	force = true
+  	force = false if directory?(new_file)
```

Comment:
It looks like `force` is only used on L560 and L562, so there's a few cases where it's not needed. Could you remove it from here so the command isn't run unless it's necessary?

You would then substitute `force ? ` with `!directory?(new_file) ? ` within the string interpolation for each. L562 would, for example be: `cmd_exec(%Q|mv #{!directory? ? "-f" : ""} "#{old_file}" "#{new_file}" && echo #{verification_token}|).include?(verification_token)`.

Reply:
okay got it, will update this :+1:

Reply:
Squashed the commits to one.